app/auth/auth_service.ts

```diff
@@ -87,6 +89,10 @@ export class AuthService {
     return requestContext;
   }
 
+  setSelectedGroupId(groupId: string) {
+    window.localStorage[SELECTED_GROUP_ID_LOCAL_STORAGE_KEY] = groupId;
```

Comment:
instead of having to refresh the whole app when a group is selected - you should just be able to pull out the logic from `register`
```
rpcService.service
      .getUser(request)
      .then((response: user.GetUserResponse) => {
        this.emitUser(this.userFromResponse(response));
      })
```
into a method called refreshUser() - where a new user will be emitted and everything should update accordingly

Reply:
Ah cool! Added that method

Reply:
i'd call refreshUser here so you can't set selected group without updating the user

alternatively you could make selectedGroup on user a getter that gets the latest state from localstorage, then you'd just need to call this.emitUser here

Reply:
Updated so it always emits a new user and also avoids reading from `localStorage` when calling `user.selectedGroup()`, since localStorage is slow